torch_em/data/datasets/monuseg.py

```diff
@@ -1,57 +1,134 @@
+    util.download_source_gdrive(zip_path, URL[split], download=download, checksum=CHECKSUM[split])
+
+    _process_monuseg(path, split)
+
+
+def _process_monuseg(path, split):
+    util.unzip(os.path.join(path, f"monuseg_{split}.zip"), path)
 
+    # assorting the images into expected dir;
+    # converting the label xml files to numpy arrays (of same dimension as input images) in the expected dir
+    root_img_save_dir = os.path.join(path, "images", split)
+    root_label_save_dir = os.path.join(path, "labels", split)
 
-# TODO
-def _process_monuseg():
-    pass
+    os.makedirs(root_img_save_dir, exist_ok=True)
+    os.makedirs(root_label_save_dir, exist_ok=True)
+
+    if split == "train":
+        all_img_dir = sorted(glob(os.path.join(path, "*", "Tissue*", "*")))
+        all_xml_label_dir = sorted(glob(os.path.join(path, "*", "Annotations", "*")))
+    else:
+        all_img_dir = sorted(glob(os.path.join(path, "MoNuSegTestData", "*.tif")))
+        all_xml_label_dir = sorted(glob(os.path.join(path, "MoNuSegTestData", "*.xml")))
+
+    assert len(all_img_dir) == len(all_xml_label_dir)
+
+    for img_path, xml_label_path in tqdm(zip(all_img_dir, all_xml_label_dir),
+                                         desc=f"Converting {split} split to the expected format",
+                                         total=len(all_img_dir)):
+        desired_label_shape = imageio.imread(img_path).shape[:-1]
+
+        img_id = os.path.split(img_path)[-1]
+        dst = os.path.join(root_img_save_dir, img_id)
+        shutil.move(src=img_path, dst=dst)
+
+        _label = util.generate_labeled_array_from_xml(shape=desired_label_shape, xml_file=xml_label_path)
+        _fileid = img_id.split(".")[0]
+        imageio.imwrite(os.path.join(root_label_save_dir, f"{_fileid}.tif"), _label)
+
+    shutil.rmtree(glob(os.path.join(path, "MoNuSeg*"))[0])
+    if split == "train":
+        shutil.rmtree(glob(os.path.join(path, "__MACOSX"))[0])
 
 
 def get_monuseg_dataset(
-    path, patch_shape, download=False, offsets=None, boundaries=False, binary=False, **kwargs
+    path, patch_shape, split, organ_type: Optional[List[str]] = None, download=False,
+    offsets=None, boundaries=False, binary=False, **kwargs
 ):
-    _download_monuseg(path, download)
+    """Dataset from https://monuseg.grand-challenge.org/Data/
+    """
+    _download_monuseg(path, download, split)
+
+    image_paths = sorted(glob(os.path.join(path, "images", split, "*")))
+    label_paths = sorted(glob(os.path.join(path, "labels", split, "*")))
+
+    if split == "train" and organ_type is not None:
+        # get all patients for multiple organ selection
+        all_organ_splits = sum([ORGAN_SPLITS[o] for o in organ_type], [])
 
-    image_path = os.path.join(path, "images")
-    label_path = os.path.join(path, "labels")
+        image_paths = [
+            _path for _path in image_paths if os.path.split(_path)[-1].split(".")[0] in all_organ_splits
```

Comment:
instead of
```python
os.path.split(_path)[-1].split(".")[0]
```
you can do
```python
Path(_path).stem
```
after doing
```python
from pathlib import Path
```
at the top. That achieves the same, but is much more readable.

Reply:
Ahha that's cool. Thanks for sharing!